pkg/api/v1/types.go

```diff
@@ -2292,6 +2292,12 @@ type Binding struct {
 	Target ObjectReference `json:"target"`
 }
 
+// Preconditions must be fulfilled before an operation (update, delete, etc.) is carried out.
+type Preconditions struct {
+	// Specifies the target UID.
+	UID *types.UID `json:"uid,omitempty"`
```

Comment:
Yes, it means only perform the operation if the named object has the given UID.

RV precondition is trivial to add later, but I don't think it's nearly as useful as a UID precondition.